### ISSUE TITLE
Nested array destructuring throws a syntax error.

### ISSUE THREAD

**Issue by aickin** [babel/babylon#462]
### Input Code

```js
require('babylon').parse(
`function test() {
  var [x, ...[y, ...z]] = [1,2,3,4];
  return x === 1 && y === 2 && z + '' === '3,4';
}`);
```

### Babylon/Babel Configuration

None, just ran the code above from a directory with no .babelrc file.

### Expected Behavior

According to [kangax's ES2016+ tests](http://kangax.github.io/compat-table/es2016plus/#test-nested_rest_destructuring,_declarations) (from which this code is taken), this should be a valid way to destructure an array, setting x to 1, y to 2, and z to [3,4].

Chrome 57/OSX and Firefox 52/OSX both allow this syntax in my testing.

### Current Behavior

Parsing this function with babylon currently throws a SyntaxError:

```
SyntaxError: Unexpected token (2:13)
    at Parser.pp$5.raise (/Users/sashaaickin/code/hermes/hermes-server/node_modules/babylon/lib/index.js:4215:13)
    at Parser.pp.unexpected (/Users/sashaaickin/code/hermes/hermes-server/node_modules/babylon/lib/index.js:1599:8)
    at Parser.pp$3.parseIdentifier (/Users/sashaaickin/code/hermes/hermes-server/node_modules/babylon/lib/index.js:4101:10)
    at Parser.pp$2.parseBindingIdentifier (/Users/sashaaickin/code/hermes/hermes-server/node_modules/babylon/lib/index.js:2872:15)
    at Parser.pp$2.parseRest (/Users/sashaaickin/code/hermes/hermes-server/node_modules/babylon/lib/index.js:2863:24)
    at Parser.pp$2.parseBindingList (/Users/sashaaickin/code/hermes/hermes-server/node_modules/babylon/lib/index.js:2913:56)
    at Parser.pp$2.parseBindingAtom (/Users/sashaaickin/code/hermes/hermes-server/node_modules/babylon/lib/index.js:2888:28)
    at Parser.pp$1.parseVarHead (/Users/sashaaickin/code/hermes/hermes-server/node_modules/babylon/lib/index.js:2192:18)
    at Parser.pp$1.parseVar (/Users/sashaaickin/code/hermes/hermes-server/node_modules/babylon/lib/index.js:2175:10)
    at Parser.pp$1.parseVarStatement (/Users/sashaaickin/code/hermes/hermes-server/node_modules/babylon/lib/index.js:2008:8)
```
### Your Environment

| software         | version
| ---------------- | -------
| Babylon          |  6.16.1
| node             | 6.9.1
| npm              | 3.10.8
| Operating System | OSX 10.12.3

**Comment by hzoo** [comment]
Hey @aickin! We really appreciate you taking the time to report an issue. The collaborators
on this project attempt to help as many people as possible, but we're a limited number of volunteers,
so it's possible this won't be addressed swiftly.

If you need any help, or just have general Babel or JavaScript questions, we have a vibrant Slack
community that typically always has someone willing to help. You can sign-up [here](https://slack.babeljs.io/)
for an invite.

**Comment by aickin** [comment]
Update: I tried `babylon@7.0.0-beta.8`, and that version parses the function just fine. `babel@7.0.0-alpha.7` also seems to do fine with this code, whereas the most current released version of babel throws an error.

**Comment by danez** [comment]
Thanks, yes this will be fixed in 7.0. I'm closing this, maybe we will backport this fix if we find the time.
